### PR TITLE
feat: remove ILogFile interface from IEditor

### PR DESCRIPTION
**Pull request by pollend**
## What does this PR do?

remove ILog interface from IEditor. I replaced these instances with AZ_Warning/AZ_Printf.

### CODE REVIEW

**Comment by pollend** [comment]
any clue about CLog and TraceLogger. There is an ebus `AZ::Debug::TraceMessageBus::Handler` for logging messages.

File: Code/Editor/Undo/Undo.cpp [hunk 1]
```diff
@@ -307,7 +306,7 @@ void CUndoManager::Redo(int numSteps)
 
     if (m_bRecording || m_bSuperRecording)
```
Comment (guillaume-haerinck) [inline Code/Editor/Undo/Undo.cpp:307]: As a side note I think o3de is using another undo system so it might be possible to nuke this file

Comment (guillaume-haerinck) [reply Code/Editor/Undo/Undo.cpp:307]: Yeah just checked we still have a few occurences in LyShine and Trackview

Comment (pollend) [reply Code/Editor/Undo/Undo.cpp:307]: They forward their state to each other. There are still a couple of places where the old info system is used. I might of deleted them already.

Comment (guillaume-haerinck) [reply Code/Editor/Undo/Undo.cpp:307]: Maybe you can add a comment at the top of this class to state that it is deprecated and should use UndoStack from Framework\AzToolsFramework\AzToolsFramework\Undo\UndoSystem.h

Comment (pollend) [reply Code/Editor/Undo/Undo.cpp:307]: there is a rfc for deprecation notice i guess the pattern is to remove something in two releases. i guess something like this doesn't need to have a notice i'm not sure ...

https://github.com/pollend/o3de/commit/45bdd38c55e94a91989be733aa68722eadef699f#diff-176a2b2d54d4b07b48d218a442ef716a5dee590007aecb10439b6f904f847cd8

Comment (pollend) [reply Code/Editor/Undo/Undo.cpp:307]: I think its probably better to track this in a github issue?

**Review by nick-l-o3de — APPROVED** [review]

**Review by gadams3 — APPROVED** [review]

**Pull request merged**